call-home/src/bin/callhome/collector/report_models.rs

```diff
@@ -116,6 +200,166 @@ impl Nexus {
     }
 }
 
+/// StorageMedia contains storage media devices count, total capacity of all the storage media
+/// devices and disk_types which contains capacity and count of storage media for each disk_type.
+#[derive(Serialize, Deserialize, Debug, Default)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct StorageMedia {
+    count: u64,
+    total_capacity_in_bytes: u64,
+    disk_types: HashMap<String, DiskTypeInfo>,
+}
+
+impl StorageMedia {
+    /// Receives a vector of openapi::models::BlockDevice and returns a new
+    /// report_models::StorageMedia object by using the data provided.
+    pub(crate) fn new(disks: Vec<BlockDevice>) -> Self {
+        let mut disk_types = HashMap::new();
+
+        // obtain the details of each disk using the Block Device API.
+        for device in disks.clone() {
+            let disk_type = get_disk_type(device.clone());
+
+            // add the capacity of the disk to the total capacity for the disk type
+            disk_types
+                .entry(disk_type.clone())
+                .or_insert(DiskTypeInfo::default())
+                .capacity_in_bytes += device.size * 512;
```

Comment:
is size always in 512B blocks, even if the device is 4k?

Reply:
Yes. Linux always considers sectors to be 512 bytes long independently of the devices real block size. According to this source.
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/linux/types.h?id=v4.4-rc6#n121